test/t8_test_cmesh_copy.c

```diff
@@ -37,40 +47,249 @@ test_cmesh_committed (t8_cmesh_t cmesh)
 }
 
+
+/*
+ *The functions t8_get_*_cmesh_testcases return the number of 
+ *testcases for a given cmesh type.
+ */
+static  int  t8_get_comm_only_cmesh_testcases():
+{
+  /*number of functions that only take comm as input * number of comm */
+  return 10*num_comm;
+}
+
+static  int  t8_get_new_hypercube_cmesh_testcases():
+{
+  /* number of element types * number of comm*3 binary variables(do_bcast, do_partition,periodic) */
+  return T8_ECLASS_COUNT*num_comm*binary*binary*binary;
+}
+
+static  int  t8_get_new_empty_cmesh_testcases():
+{
+  /* number of comm* 1 binary variable(do_partition)* possible dimensions(check dim 0 to 4, therefore we need to add 2 to dim)*/
+  return num_comm*binary*(dim+2);
+}
+
+static  int  t8_get_new_from_class_cmesh_testcases():
+{
+  /* number of element types * number of comm */
+  return T8_ECLASS_COUNT*num_comm;
+}
+
+static  int  t8_get_new_hypercube_hybrid_cmesh_testcases():
+{
+  /* possible dim * number of comm*2 binary variables(do_partition,periodic) */
+  return 4*num_comm*binary*binary;
+}
+
+static  int t8_get_new_periodic_cmesh_testcases():
+{
+  /*  number of comm * possible dim */
+  return num_comm*dim;
+}
+
+static  int  t8_get_new_bigmesh_cmesh_testcases():
+{
+  /*  number of element types * number of trees * number of comm */
+  return T8_ECLASS_COUNT*max_num_of_trees*num_comm;
+}
+
+static  int  t8_get_new_prism_cake_cmesh_testcases():
+{
+  /*  number of comm * number of prisms */
+  return num_comm*max_num_of_prisms;
+}
+
+static  int t8_get_new_disjoint_bricks_cmesh_testcases():
+{
+  /*  TO DO: add comment*/
+  return 20*20*20*binary*binary*binary*num_comm;
+}
+
+static int t8_get_all_testcases():
+{
+  /* The number of all tests.*/
+  return t8_get_comm_only_cmesh_testcases()+t8_get_new_hypercube_cmesh_testcases()+t8_get_new_empty_cmesh_testcases()+t8_get_new_from_class_cmesh_testcases()+t8_get_new_hypercube_hybrid_cmesh_testcases()+t8_get_new_periodic_cmesh_testcases()+t8_get_new_bigmesh_cmesh_testcases()+t8_get_new_prism_cake_cmesh_testcases()+t8_get_new_disjoint_bricks_cmesh_testcases();
+}
+
+
+static              t8_cmesh_t
+t8_test_create_comm_only_cmesh(int cmesh_id)
+{
+  switch ((int)(cmesh_id/3)) {
+  case 0:
+    return t8_cmesh_new_periodic_tri (comm[cmesh_id % 3]);
+  case 1:
+    return t8_cmesh_new_periodic_hybrid (comm[cmesh_id % 3]);
+  case 2:
+    return t8_cmesh_new_periodic_line_more_trees (comm[cmesh_id % 3]);
+  case 3:
+    return t8_cmesh_new_line_zigzag (comm[cmesh_id % 3]);
+  case 4:
+    return t8_cmesh_new_prism_deformed (comm[cmesh_id % 3]);
+  case 5:
+    return t8_cmesh_new_prism_cake_funny_oriented (comm[cmesh_id % 3]);
+  case 6:
+    return t8_cmesh_new_prism_geometry (comm[cmesh_id % 3]);
+  case 7:
+    return t8_cmesh_new_tet_orientation_test (comm[cmesh_id % 3]);
+  case 8:
+    return t8_cmesh_new_hybrid_gate (comm[cmesh_id % 3]);
+  case 9:
+    return t8_cmesh_new_hybrid_gate_deformed (comm[cmesh_id % 3]);
+  }
+}
+
```

Comment:
Also here and for the other functions:

A brief comment what this function does.
Also for each input variable what does it do?
Have a look at some .h files in t8code (for example t8_forest.h) on how we document functions and parameters.